### ISSUE TITLE
trouble with H bond from two different chains

### ISSUE THREAD

**Issue by Cold7** [MDAnalysis/mdanalysis#975]
### Expected behaviour

To get intra and inter chain H bonds in a PDB file  (pdb ID [2bg9](http://www.rcsb.org/pdb/explore/explore.do?structureId=2BG9)  ). For example using PDB2PQR programm to get HBond, it found "Donor: TYR A 127 OH	Acceptor: ASN E 38 OD1".

Edit: [Here is the PDB with H atoms ](https://dl.dropboxusercontent.com/u/49250897/asdf.pdb)

### Actual behaviour

Only intra chain H bonds are computed (only H bonds from the same chain)

### Code to reproduce the behaviour
``` python
import MDAnalysis as mda
import MDAnalysis.analysis.hbonds
u=MDAnalysis.Universe("2bg9.pdb")
h = MDAnalysis.analysis.hbonds.HydrogenBondAnalysis(u, 'protein', 'protein', distance=3.0, angle=120)		
h.run()



### Currently version of MDAnalysis:
 0.15.0

**Comment by orbeckst** [comment]
2BG9 does not have hydrogens. The H-bond analysis needs hydrogen atoms to calculate angles. Add hydrogens first.

**Comment by Cold7** [comment]
I forgot to say that I add hydrogen atoms through PDB2PQR too, [here you can download the structure (link to dropbox)](https://dl.dropboxusercontent.com/u/49250897/asdf.pdb)

**Comment by orbeckst** [comment]
With your file, I get 3950 h-bonds:
```python
>>> import MDAnalysis as mda
>>> import MDAnalysis.analysis.hbonds as hbonds
>>> print mda.__version__
0.15.1-dev0
>>> u = mda.Universe("./asdf.pdb")
>>> h = hbonds.HydrogenBondAnalysis(u, "protein", "protein")
>>> h.run()
HBonds frame     1/1 [100.0%]
>>> print len(h.timeseries[0])
3950
>>> h.generate_table()
>>> print len(h.table)
3950
```

Typically, I find the `h.table` (a recarray) more convenient to work with but you can either unpack the timeseries structure or use the recarray.

I used the latest develop version of MDAnalysis. When you try the above commands please also report your version.

**Comment by orbeckst** [comment]
By the way, I get exactly the same result (3950 h-bonds) with 
```python
h = hbonds.HydrogenBondAnalysis(u, "protein", "protein", distance=3.0, angle=120)
```

(and I just saw that you are using the latest stable 0.15.0.)

**Comment by orbeckst** [comment]
... and I also get 3950 with MDAnalysis 0.15.0.

@Cold7 can you please double check that the code as I wrote it also works for you?

**Comment by orbeckst** [comment]
I am re-reading your issue report and realize that I misunderstood. You get H-bonds but you do not see the *inter*-chain ones.

To make this a bit easier, can you instruct PDB2PQR to keep chain identifiers? The asdf.pdb is loaded with all atoms `len(u.atoms) == 29879` but MDAnalysis gets confused about the residues (`len(u.atoms.residues) == 431` – that's arguably a bug that might require its own issue report). Alternatively, if you know the first and last atom in each chain you can add the chain identifiers in MDAnalysis (subtract 1 to go from continuous PDB indices to atom array indices):
```python
import numpy as np
chains = "ABCDE"
# I manually looked up the atom indices that start chains in asdf.pdb
starts = np.array([1, 5999, 11927, 17899, 23913]) - 1  # subtract 1 for MDA atom index
stops = np.concatenate([starts[1:] + 1, [None]])

for chain, start, stop in zip(chains, starts, stops):
    u.atoms[start:stop].segids = chain
```

We can now do lookups on the chains using the atom indices in `h.table`:
```python
donor_chains = u.atoms[h.table.donor_index].segids
acceptor_chains = u.atoms[h.table.acceptor_index].segids

print(all(donor_chains == acceptor_chains))
# False, i.e., there are some cross chain h-bonds

# get the h-bonds across chains (i.e. "not same segid")
inter_h_bonds = h.table[~(donor_chains == acceptor_chains)]
print(len(inter_h_bonds))
# 23
```

From the above I conclude that there are a few inter-chain H-bonds. Admittedly, 23 does not sound like a big number (only 4.6 per interface). But can you verify that these H-bonds are correct?

```python
>>> h.table[~(donor_chains == acceptor_chains)]
rec.array([ (0.0, 23940, 23900, 23939, 23899, u'GLU', 3, u'H', u'ASN', 1, u'O', 2.169037103652954, 174.46319580078125),
 (0.0, 18594, 24658, 18593, 24657, u'ASN', 42, u'HD22', u'GLU', 46, u'OE1', 2.4219658374786377, 148.05714416503906),
 (0.0, 7260, 14488, 7259, 14487, u'SER', 79, u'H', u'GLU', 157, u'OE1', 2.8851985931396484, 157.56292724609375),
 (0.0, 19659, 26300, 19658, 26299, u'LYS', 107, u'HZ3', u'GLN', 148, u'O', 2.617757797241211, 138.61746215820312),
 (0.0, 2105, 24528, 2104, 24527, u'TYR', 127, u'HH', u'ASN', 38, u'OD1', 2.077479600906372, 155.39517211914062),
 (0.0, 21835, 28797, 21834, 28796, u'SER', 239, u'HG', u'LEU', 308, u'O', 2.414135217666626, 126.5952377319336),
 (0.0, 3975, 10865, 3974, 10864, u'LYS', 242, u'HZ3', u'HIS', 312, u'NE2', 2.3941643238067627, 165.59437561035156),
 (0.0, 23940, 23900, 23939, 23899, u'GLU', 3, u'H', u'ASN', 1, u'O', 2.169037103652954, 147.27110290527344),
 (0.0, 7260, 14488, 7259, 14487, u'SER', 79, u'H', u'GLU', 157, u'OE1', 2.8851985931396484, 128.59280395507812),
 (0.0, 9384, 16220, 9383, 16219, u'TYR', 220, u'HH', u'PRO', 279, u'O', 2.344447135925293, 133.4222869873047),
 (0.0, 13301, 20385, 13300, 20384, u'ARG', 83, u'HH22', u'TYR', 151, u'OH', 2.2233614921569824, 156.27638244628906),
 (0.0, 15821, 22904, 15820, 22903, u'SER', 253, u'HG', u'HIS', 306, u'ND1', 2.61795711517334, 167.5908966064453),
 (0.0, 15907, 21902, 15906, 21901, u'THR', 259, u'HG1', u'THR', 244, u'OG1', 1.2270634174346924, 149.4663543701172),
 (0.0, 19230, 26397, 19229, 26396, u'ARG', 79, u'HH12', u'GLU', 154, u'OE1', 1.9945298433303833, 131.55413818359375),
 (0.0, 19230, 26398, 19229, 26397, u'ARG', 79, u'HH12', u'GLU', 154, u'OE2', 1.6676945686340332, 149.3871307373047),
 (0.0, 19225, 26398, 19224, 26397, u'ARG', 79, u'HH22', u'GLU', 154, u'OE2', 2.0408272743225098, 134.31906127929688),
 (0.0, 19659, 26300, 19658, 26299, u'LYS', 107, u'HZ3', u'GLN', 148, u'O', 2.617757797241211, 134.27586364746094),
 (0.0, 19660, 26319, 19659, 26318, u'LYS', 107, u'HZ2', u'THR', 149, u'OG1', 2.0901763439178467, 143.83973693847656),
 (0.0, 23911, 25021, 23910, 25020, u'ASN', 1, u'HD22', u'ASN', 67, u'OD1', 2.7247841358184814, 144.9536895751953),
 (0.0, 23912, 25044, 23911, 25043, u'ASN', 1, u'HD21', u'SER', 69, u'OG', 2.891068696975708, 126.60828399658203),
 (0.0, 25189, 2503, 25188, 2502, u'ARG', 78, u'HH22', u'ASP', 152, u'OD1', 1.9085960388183594, 155.13197326660156),
 (0.0, 7260, 14488, 7259, 14487, u'SER', 79, u'H', u'GLU', 157, u'OE1', 2.8851985931396484, 146.90914916992188),
 (0.0, 27804, 4866, 27803, 4865, u'GLN', 245, u'HE21', u'ASN', 297, u'OD1', 2.205596685409546, 126.63280487060547)],
          dtype=[('time', '<f8'), ('donor_idx', '<i8'), ('acceptor_idx', '<i8'), ('donor_index', '<i8'), ('acceptor_index', '<i8'), ('donor_resnm', '<U4'), ('donor_resid', '<i8'), ('donor_atom', '<U4'), ('acceptor_resnm', '<U4'), ('acceptor_resid', '<i8'), ('acceptor_atom', '<U4'), ('distance', '<f8'), ('angle', '<f8')])
```

**Comment by Cold7** [comment]
thanks @orbeckst it was very helpful (and now I think that h.table is a faster way to see hbond and more easily that unpack the structure). Using this code I checked that exist interchain H bonds (and are the same that PDB2PQR gave as output), but using mdanalysis with other pdb (with H atoms) I can not see it and now I will look what happend in the other case (I tested with others threshold values and it did not appear). Well, now I know that is possible to compute interchain H bonds using mdanalysis. 

Thanks again and best wishes
